app/controllers/questions_controller.rb

```diff
@@ -15,40 +15,57 @@ def index
     render json: questions
   end
 
+  def user_questions
+    questions = @current_user.questions
+    if questions.empty?
+      render json: [], status: :ok
+    else
+      render json: questions, status: :ok
+    end
+  end
+
   def show
-    question = Question.find(params[:id])
-    render json: question
+    answer = @question.answers
+    answer_array = Array(answer)
+    render json: { question: @question, answer: answer_array }, status: :ok
```

Comment:
1. @question.answers returns an array of answers so why are you typecasting it again to Array. 
2. This show function returns all the answers, but it doesn't return the name of the user who wrote the answers.

The sample output of this api should be:

```
{
  question: "Sample Question here",
  answer: [
    {
      "description": "Sample Answer 1",
      "written_by": "John Doe",
      "created_at": "xyz time"
    }.
    {
      "description": "Sample Answer 2",
      "written_by": "Alice",
      "created_at": "abc time",
    },
  ],
}.
```